### ISSUE TITLE
Google as an Auth provider doesn't work on iOS saved to homescreen

### ISSUE THREAD

**Issue by Haroenv** [firebase/firebase-js-sdk#77]
### [REQUIRED] Describe your environment

  * Operating System version: iOS 10
  * Firebase SDK version: 4.1.3
  * Firebase Product: auth

#### Steps to reproduce:

0. Save the app on the homescreen on iOS
1. make a redirect or popup login with google Auth provider
2. if redirect:
   - webview is not a "valid user agent"
3. if popup:
    - gets stuck in Safari about:blank, doesn't open the app again to pass on the token

#### Relevant Code:

```javascript
import firebase from "firebase";

const config = {
//...
};

firebase.initializeApp(config);

const googleAuthProvider = new firebaseAuth.GoogleAuthProvider();

firebase.auth().signInWithRedirect(googleAuthProvider);
// or
firebase.auth().signInWithRedirect(googleAuthProvider);
```

**Comment by Haroenv** [comment]
related issue: https://github.com/firebase/quickstart-js/issues/94

**Comment by bojeil-google** [comment]
This is a known issue and not really Firebase related (in the case of redirect) which is Google sign-in related. Webviews are insecure and Google sign-in no longer works in that environment anymore. Google sign-in recommends using SFSafariViewController for this but homescreen apps cannot run in a SFSafariViewController, instead they always run in a webview environment.
As for popups, they won't work for a different reason since Firebase OAuth popup relies on web storage to pass the result back to the app. However homescreen apps do not share the same web storage as the system browser (which is how the popup is opened).

**Comment by Haroenv** [comment]
I totally understand that, but saving to home screen isn’t something I have control over as an author. Does this really come down to “it’s impossible”?

**Comment by Bretto** [comment]
This is a pretty big, deal breaker, any hope/news on this ?

**Comment by bojeil-google** [comment]
Sorry, it is out of our control. You need to reach out to the Google OAuth folks about this. I don't think there is currently anyway to get Google sign-in to work in a home screen iOS app.

**Comment by RileyDavidson-Evans** [comment]
Android is also effected by this issue, this is definitely a deal breaker for pwa

**Comment by zywyz** [comment]
There are still no solution for this problem?

**Comment by cdharris** [comment]
Have just run into this issue myself, some of our users are trying to launch from iOS bookmarks and are unable to sign in. I appreciate this is not a firebase issue per say but wanted to add my voice here to draw attention that there are more affected users. 

I am using firebase web authentication, is there no way to allow login from iOS in this instance?

Thanks

**Comment by natezerk** [comment]
This makes me sad.

**Comment by natezerk** [comment]
Facebook login doesn't seem to work either?

**Comment by bojeil-google** [comment]
Facebook login and other non-Google providers should work in redirect mode only. 
As I mentioned earlier, it is impossible to get Google sign-in in that iOS homescreen mode. From the Google sign-in perspective, it is coming from an embedded webview which are now blocked.

**Comment by Bretto** [comment]
Does anyone know where we can follow the progress on this PWA deal breaker ?

**Comment by mikeatmarvel** [comment]
We would welcome a solution to this problem as well.  Unfortunately this is a "deal breaker" for us as well....  I wanted to share on this thread as we are running into this same issue and did not want to be silent about it.  

Looking forward to a resolution soon!

**Comment by ajonp** [comment]
How do we upvote a solution?

**Comment by mesqueeb** [comment]
Does anyone know where we can follow information on how this problem is being treated by either Google OAuth, Apple or Firebase? 

Is there a way for firebase to detect if the app is launched from the home screen so we can display an explanation for our users?

**Comment by bojeil-google** [comment]
The iOS embedded webview has a distinct useragent. You can test for that and display a message to the user.  Embedded webviews for iOS 9 and above are blocked from accessing the Google OAuth page for security reasons: https://developers.googleblog.com/2016/08/modernizing-oauth-interactions-in-native-apps.html

**Comment by tcha-tcho** [comment]
Why not just:

```
  var fbAppId = 'XXXXXXXXXXXXXX';
  var fburl = "https://www.facebook.com/dialog/oauth?";
  fburl += "client_id="+fbAppId;
  fburl += "&redirect_uri="+window.location.href;
  window.location.href = fburl;
```

When popup is not possible, like in Safari?

**Comment by henrylearn2rock** [comment]
https://bugs.chromium.org/p/chromium/issues/detail?id=771418 seems to be fixed in Chrome dev  64.0.3254.2

**Comment by nicolasgarnier** [comment]
For me it looks like redirect auth works on https://friendly-pix.com/ on iOS homescreen apps with Google sign-in. But perhaps, since this is an old app, I was grandfathered into an exception.

Popup auth works on Android Homescreen apps.

This is how I implemented an exception just for iOS homescreen apps:


```js
  var signInFlow = 'popup';
  // For iOS full screen apps we use the redirect auth mode.
  if (('standalone' in window.navigator)
      && window.navigator.standalone){
    signInFlow = 'redirect';
  }

  // FirebaseUI config.
  var uiConfig = {
    'signInFlow': signInFlow,
    'signInOptions': [
      firebase.auth.GoogleAuthProvider.PROVIDER_ID,
      firebase.auth.FacebookAuthProvider.PROVIDER_ID
    ]
  };
  var firebaseUi = new firebaseui.auth.AuthUI(firebase.auth());
```

**Comment by jcyh0120** [comment]
Any update for this feature? I want to give up using PWA because my user cannot login.

**Comment by RileyDavidson-Evans** [comment]
This may have changed but about 6 months ago when I was attempting this I discovered apple has deemed WebViews insecure and thus, secure login methods through PWA's by design will not work.

**Comment by mesqueeb** [comment]
Today this issue is fixed to my findings! 
Does anyone still experience the inability to log in?

**Comment by jcyh0120** [comment]
How do you solve this? @mesqueeb

**Comment by mesqueeb** [comment]
@jcyh0120 
It suddenly works for my app!
The google auth login in my app at https://listo.lucaban.com works now even within the safari webview.
You can try it out by adding it on the homescreen and login.

**Comment by jcyh0120** [comment]
@mesqueeb What kind of oauth did you user? SigninWith Popup or SignInWithRedirect?

**Comment by mesqueeb** [comment]
@jcyh0120 I use redirect.

**Comment by jcyh0120** [comment]
I got this error with my last build using login with redirect.

https://user-images.githubusercontent.com/29078867/27479671-c5b7c88a-580c-11e7-9df8-be1016e2b106.PNG

**Comment by mesqueeb** [comment]
@jcyh0120 That's weird. I don't get that bug at all. I can log in without problems.
Maybe it's iOS version related? I'm on iOS 11.2.5

**Comment by jcyh0120** [comment]
@mesqueeb I use ios 9.x.x and ios 10.x.x. Both can not do Google OAuth with Add to home screen.

**Comment by johnozbay** [comment]
We're developing a pinned-home-screen mobile web app, and now have the same issue.
@mesqueeb and @jcyh0120 you're both correct. 

I started testing different iOS versions to try to isolate the versions impacted by this. 
So far what I've found is : 

**iOS v9.3.5** - Not Working _(disallowed_user_agent error)_
**iOS v10.3.2** - Not Working _(disallowed_user_agent error)_
**iOS v11.2** - Working
**iOS v11.2.5** - Working

Also, we use the redirect flow for pinned web apps and the pop up flow for non-web-apps by using this to detect the state : 

```
var isInWebAppiOS = (window.navigator.standalone === true);
var isInWebAppChrome = (window.matchMedia('(display-mode: standalone)').matches);
```

So far the only possible workaround I can think of is:  
1) detect OS version, 
2) (if we know we'll get the disallowed_user_agent error) ~ authenticate user in the browser, 
3) grab all auth related tokens and info, 
4) append tokens and info into the url, 
5) then prompt user to add web app to home screen.
6) once in the pinned web app, check URL for token, and restore auth session using that. 

^^ Haven't tested this yet ~ open for suggestions / workarounds / ideas / memes / curses anything.

**Comment by johnozbay** [comment]
I think I solved the auth problem for ios home screen web applications entirely, and it seems to work for all iOS Devices & Versions. 

**TLDR;**
[Here is the github gist for the code](https://gist.github.com/johnozbay/94c0fcf6af7b6dba8808430bfc726fc1) :)

**Solution**
First, you'll need to lead your users to login in Safari, **then** add your webapp to their home screen.

Once your users are logged in, and you got your user object in `firebase.auth().onAuthStateChanged` Get everything from localStorage, convert it to a base64 string and append it to the end of the URL with a parameter using `window.history.pushState`.

When your users pin your website to their homescreen, check this parameter, and if there's a base64 version of the localStorage in the URL, set the localStorage from the base64 and reload the page with `window.location.reload(false)`.

This tactic basically allows you to port everything from Safari's localStorage to the pinned web app's local storage using URL parameters.

[Made a gist for the code](https://gist.github.com/johnozbay/94c0fcf6af7b6dba8808430bfc726fc1)

**Comment by bojeil-google** [comment]
Please do not rely on hacking `localStorage`. It is an implementation detail which we have the right to change. We have plans to migrate from it to `indexedDB`.
In addition, I would not recommend keeping your Auth state (which contains the indefinite refresh token, opening a large window of attack) in the URL parameters at all times hoping a user would add your app to the homescreen.

**Comment by johnozbay** [comment]
Thanks for the quick response @bojeil-google! and I appreciate the heads up. ✌🏻

For my personal education, two questions : 

1) Wouldn't appending the token to the URL only using javascript (and not involving any servers or anything, purely on the client side with `pushState`) be as safe as localStorage? They're both only accessible on the client side via javascript, and if one has a way of executing any malicious code, wouldn't this be as bad for keeping things in localStorage too? 

2) Is there anything else we can use/do to move a temporary token to the web app, through the URL to achieve the same results in a less hacky way? 

In my specific case it's not _hoping_ the user would add the app to the home screen. 😕 It's the expected behavior, so I need a way around this if I don't want to exclude all users between iOS9 up to iOS11.

**Comment by nicolasgarnier** [comment]
@bojeil-google I'm guessing the only way to truly fix this is to work on fixing the popup flow to work on iOS homescreen apps. Not saying it's easy or even possible but if there is any way to communicate between the popup and the app that works it might be worth a hack... (I know, no-one likes hacks...)

**Comment by bojeil-google** [comment]
Hey @nicolasgarnier, I don't know of any way to pass the result securely enough that we can provide as a built in solution in our SDK. The issue with embedded webviews (iOS homescreen apps use those) is that they are insecure. The user has no way to tell which website they are signing in to as there is no indicator of the URL identifying the page they are entering their credentials to. Ideally, this should be implemented with SFSafariViewController or SFAuthenticationSession but iOS does not support those in homescreen apps. Chrome homescreen apps on the other hand, support Chrome custom tabs which have same security as SFSafariViewController and SFAuthenticationSession and even share same storage as the homescreen app itself.

Regarding @johnozbay's suggestions:
From my understanding, you don't know if the user will actually add your app to homescreen. This action has to be taken outside the web app which means you have to keep the Auth state in the URL the whole time. If this is a one time operation and you know when it will happen, you could try to pass the ID token instead and then exchange it for a custom token in your homescreen app (detect it, send it to server, verify it, mint custom token and return it to sign in with ) after you verify the `auth_time` in the ID token is recent (user recently signed in) to minimize token theft risks.

Also if you are appending the auth state in the URL, it will end up getting logged on your server, other web analytics tracking tools, etc. Embedded in that link is the user's Firebase refresh token. So if these logs are compromised, the refresh tokens of all these users could be decoded and stolen no matter how old they are (refresh tokens are indefinite unless revoked).

**Comment by johnozbay** [comment]
Thanks a million @bojeil-google for taking the time for the detailed explanation. I now have a much clearer understanding of what's going on, and Google Auth's perspective on why WebViews are insecure. (regarding the lack of URL)

In my specific case, we're deploying a cross-platform mobile web-app, and it's intended specifically to be used on mobile devices as a pinned app, and we're directing users to do so with specific UX. It is a one time operation and we can even direct the users to make it on a specific page.

I'll pass the ID token and do the custom token exchange & verify `auth_time` instead 👍🏻

Plus side of using Firebase is that we have no servers, nor any analytics or tracking tools. Static HTML hosted on Firebase Hosting, and some Cloud Functions here and there. So there should be no logs in our specific case.

Again many thanks for the help & clarifications.

**Comment by mesqueeb** [comment]
Dear @bojeil-google 
You said:
> Ideally, this should be implemented with SFSafariViewController or SFAuthenticationSession but iOS does not support those in homescreen apps.

But since the problem does not occur anymore after iOS v11.2, do you think Apple already changed that behaviour?

And also, should we change this issue from:
_Google as an Auth provider doesn't work on iOS saved to homescreen_
to:
_Google as an Auth provider doesn't work on iOS saved to homescreen **before iOS v11.2**_?

**Comment by bojeil-google** [comment]
Hey @mesqueeb,  I don't know the answer to that. iOS v11.2 is fairly new. I haven't had a chance to test it yet. Will try to get my hands on a device running that version of iOS and test it.

**Comment by henrylearn2rock** [comment]
@bojeil-google https://github.com/google/google-api-javascript-client/issues/260 Looks like even google-api-javascript-client is using SessionStorage?

**Comment by henrylearn2rock** [comment]
I may be wrong but looks like google-api-javascript-client uses SessionStorage https://github.com/google/google-api-javascript-client/issues/260

https://pwa-directory.appspot.com/ This PWA uses google-api-javascript-client and google login works as expected in Android.  I'm not sure about iOS.

**Comment by bojeil-google** [comment]
I am not sure what you are trying to say. This issue is specific to iOS. I also have no idea why it matters that they use sessionStorage. We use sessionStorage too for redirect mode.

**Comment by rmmmp** [comment]
Not sure if this can be considered as related but Google sign in still doesn't work in Safari Webview for apps like Facebook and Facebook Messenger even in iOS 11.2.

I mentioned it here because I think those have the same setup as a homescreen app.

**Comment by mesqueeb** [comment]
Then I must warn my users if I share my app through facebook with something like:
"It seems the app is open inside Facebook. Please open it in Safari."

@rmmmp Do you know how recognise with JS if it's inside Facebook messenger or not?

**Comment by rmmmp** [comment]
@mesqueeb - I'm not sure how to specifically detect if in Facebook. But you can detect if it's in Safari Webview mode. See [here](https://stackoverflow.com/a/9951404).

**Comment by JaySunSyn** [comment]
@johnozbay does iOS v11.3 work for you? (does not for me)

**Comment by dyegolara** [comment]
it does not work for me on iOS 11.3 with redirect nor pop-up

**Comment by johnozbay** [comment]
Unfortunately it doesn't @JaySunSyn. 🙁 All that being said, I absolutely understand google's point of view on this issue. It is an inherent security risk, and it's something either Apple or we should take action on. I wouldn't want my Google creds stolen either. Looks like Apple brought PWA support in iOS 11.3 ~ so it's only more of a reason for them to fix.

To work around this I've implemented two workarounds based on the conversations up in this chain. Neither are _good solutions_, but depending on your situation, they could solve your problem. First one is all local on the device, and can be broken if Apple decides to change things up. Second involves setting up a cloud function chain.

**Solution 1)** Create a controlled flow with good design & UX and make sure the user can only "Add to Home Screen" after they're logged in to your service in Safari, auth-tokenize the URL there (with a 5min expiry timing) only on a single specific page, and ask the users to add to home screen & launch immediately. This relies on iOS not stripping URL parameters when adding to home screen, and can be broken if Apple changes this. For one specific use case, this worked quite well for me. However it still allows for an auth-tokened URL to exist for 5 minutes, hence the danger and second solution.

**Solution 2)** This one relies on the fact that you can open URLs in Safari from pinned home screen applications, but not the other way around. Flow goes like this :

_In Pinned App_
User presses sign in / sign up with Google in your pinned home screen app, 
Ping a cloud function to get a UUID, 
Save this UUID to localstorage,
then open a URL in Safari like : myapp.com/auth?uuid=12345 

_Now In Safari_
The Safari auth page gets the UUID from the URL parameter, 
saves UUID to localstorage temporarily, 
Then triggers Google's redirect auth, prompts user to login using Google. 
After a successful signup/signin, you got the ID token, 
Safari sends this token & UUID it had in localstorage to a cloud function 
Cloud Function saves the token & UUID & token creation time to your db temporarily. 
Once token is successfully saved to DB, and function returns a happy 200 status code, 
Safari prompts the user to go back to the pinned app ~ we did something like this :
 
<img width="480" alt="screen shot 2018-04-16 at 20 01 08" src="https://user-images.githubusercontent.com/611338/38824199-2189f424-41b1-11e8-8939-656036645c8b.png">

Once back in the pinned app, the app checks localstorage for a UUID,
If there's one, pings the cloud function with the UUID. 
Cloud Function delivers the ID token (and deletes both UUID & Token immediately) 
Finally, you call signinUsingToken with this token in the pinned web app, and open up a bottle of champagne. 

I'd say this is a slightly better of a solution than the first one. Takes more steps, but feels much more safer. 

I'd be happy and curious to hear the google team's thoughts on this as well. ✌🏻

**Comment by johnozbay** [comment]
@socceroos the solution proposed at the url relies on window.open() which in iOS pinned home screen application results in navigating away from your current page (so same behavior as redirect). And this issue is specific to Google's auth flow, which is different than Spotify's. 

The first post in this issue talks about the results of both redirects and popups. Shortly, 

If you do a plain redirect, you'll get a Webview is not a "valid user agent" error due to google's restrictions (Which are btw perfectly legitimate and understandable. Highly recommend that you read the reasons in this thread). 

If you use a popup: you'll get stuck in Safari about:blank, and won't open the app again to pass on the token, due to Firebase OAuth popup relying on web storage to pass the result back to the app and homescreen apps not sharing the same web storage as the system browser.

The only OS & browser version agnostic workaround I could find is the one I listed above, which is to instruct the user to press the "< Back to App" button on the top left in iOS Safari and handle the rest with some cloud functions sorcery.

**Comment by socceroos** [comment]
@johnozbay I was more thinking with the solution above that you open a popup from your PWA that is the login page (happening in Safari). OAuth occurs in Safari login page and upon completion the login page calls `window.opener.postMessage` with the token. This is received by the PWA which then does token login. Technically you'd be logging them in twice, but it would be less UX hassle.
Maybe I'm missing something?

**Comment by johnozbay** [comment]
@socceroos iOS reloads pinned home screen applications when you navigate away and back, losing its state. (and completely restarts them) As far as I know, pinned applications in iOS are 100% air-gapped with no way to pass data between pinned application & Safari reliably across all versions of iOS and Safari. (I'd love to be proven wrong here) So, sadly `postMessage` wouldn't be able to communicate back.

**Comment by abelson** [comment]
@johnozbay can you kindly share your code to the second solution? Thanks

**Comment by johnozbay** [comment]
@abelson I'm afraid not. It's deeply integrated into the login flow for a client project that supports multiple odd providers, so it would require me to rewrite a simplified version of this from scratch.

If there's any specific part that you're stuck with I'd be happy to help clarify as much as I can.

**Comment by turadg** [comment]
Fixing this may depend on https://bugs.webkit.org/show_bug.cgi?id=185400

**Comment by tomodian** [comment]
Curious that iOS Safari on iPhone (11.4) doesn't work, however iPad (11.2.5) works without any problem.

**Comment by socceroos** [comment]
@tomodian The change in browser context handling on iOS was introduced in 11.3. 11.2 is unaffected. 👍

**Comment by jorge-valvert** [comment]
I´m assuming that you have a "manifest" meta tag to reference manifest.json file.

I suggest to do this workaround, that works perfect in ios safari web app:

https://stackoverflow.com/a/52286655/8390589

**Comment by danielruiz7** [comment]
@jorge-valvert is right. Tested in my own project and social login is working now. Thanks for sharing.

**Comment by aSegatto** [comment]
@jorge-valvert solution works on iOS also for us, breaks Chrome so it still needs some imprevements

**Comment by danielruiz7** [comment]
@aSegatto for me is working both. What I did was adding two link tags for the same manifest:

<link rel="manifest" href="/manifest.json">
<link rel="pwa-setup" href="/manifest.json">

When page is loaded I check if it's on iOS and do the following in order to avoid safari to read the rel="manifest":

document.querySelector('link[rel="manifest"]').setAttribute("rel", "no-on-ios");

I'm not sure if it's necessary though.

This way Chrome is able to read the manifest via rel="manifest" and pwacompat.js is able to prepare the meta tags for iOS using rel="pwa-setup".

**Comment by jvitor83** [comment]
With:

The `https://github.com/GoogleChromeLabs/pwacompat/blob/master/pwacompat.js` modified at https://github.com/GoogleChromeLabs/pwacompat/blob/master/pwacompat.js#L36 to `const manifestEl = document.head.querySelector('link[rel="pwa-setup"]');`

and this configuration at index.html:
```
  <link rel="manifest" href="manifest.json">
  <link rel="pwa-setup" href="manifest.json">
  <script src="pwacompat.js"></script>
  <script>
   var iOS = !!navigator.platform && /iPhone|iPod/.test(navigator.platform);
   if(iOS) {
      document.querySelector('link[rel="manifest"]').setAttribute("rel", "no-on-ios");
   }
  </script>
```

it works!

**Comment by rodolfocop** [comment]
We are having a problem similar to this in our application.
If you want you can use the following code above that my friend @jvitor83  created.
Thank you for your help @danielruiz7 .

**Comment by theigl** [comment]
This works for me but without the manifest all the internal links in my application open a browser window. Did anyone run into this issue and resolve it?

**Comment by rodolfocop** [comment]
@theigl You have tried using code the @jvitor83 ? 
I am saying because this problem not occur to us.

**Comment by satoshi-m8a** [comment]
Is this article helpful? 
https://medium.com/@jonnykalambay/progressive-web-apps-with-oauth-dont-repeat-my-mistake-16a4063ce113

**Comment by robert-king** [comment]
@jvitor83 the modified pwacompat.js  solution stopped working for me recently. Was working a couple of weeks ago. Not sure what changed, perhaps it's because I updated to the latest version of angular and firebase.
Is anyone else having issues with pwacompat? I'm just going to not use google login for PWAs, seems like PWAs aren't there yet.

**Comment by joewoodhouse** [comment]
@jvitor83 thanks for your solution, a few comments/thoughts

Looking at the pwacompat.js script, the `const manifestEl` is just used to get the URL of the manifest.json file. So if you're going to modify the script, you may as well just hardcode the path to the manifest file - this would remove the need for having the 2 `<link>` tags I think? i.e.

```html
 <link rel="manifest" href="manifest.json">
 <script src="pwacompat.js"></script>
 <script>
   var iOS = !!navigator.platform && /iPhone|iPod|iPad/.test(navigator.platform) && !window.MSStream;
   if(iOS) {
      document.querySelector('link[rel="manifest"]').setAttribute("rel", "no-on-ios");
   }
 </script>
```

```js
  // pwacompat.js around line 36
  function setup() {
    const manifestHref = 'manifest.json';
    const hrefFactory = buildHrefFactory([manifestHref, window.location]);
    ...
```

Note I also tweaked the iOS detection to include iPads and to disregard some versions of IE that tried to be iPhones.

Whilst I'm pleased that the fix is getting my app working, has anyone actually worked out how/why this fixes the OAuth issue? From reading the script, all it does is read your manifest.json file and then add some `meta` tags to the page accordingly (plus some other bits with splashes and icons)- I'd be interested to know which of those tags actually makes the difference, then potentially I could remove the use of the script. Anyone have any ideas?

**Comment by jellevdvliet** [comment]
Did this "fix" stop working?

- I have both Facebook and Google auth providers. They both are with a redirect. 
- I added the pwacompat file, and used this code to "fix the bug": https://github.com/firebase/firebase-js-sdk/issues/77#issuecomment-439324680 

When I save the PWA to homescreen and try to log in, it opens Safari, I go through the oauth2 process and am redirected to the app in the Safari browser - not in the installed PWA. When I go back to the installed PWA it's still not authed.

Tested on iPhoneX with iOS 12.1.2

**Comment by aSegatto** [comment]
We are currently using the solution proposed in[comment]( https://github.com/firebase/firebase-js-sdk/issues/77#issuecomment-430565880) and it's still working.

**Comment by PutziSan** [comment]
Will Google Auth work with the next release of iOS 12.2? (Mike Hartington's [comment on twitter](https://twitter.com/mhartington/status/1089293403089784832) seems to suggest this)

**Comment by powerspowers** [comment]
I just tried the solution and while it lets you start the login to your google account you cannot complete it if Google decides to send you a text message code to confirm. When you leave to get your code and then come back the page for typing in your code disappears and you have to start the sign in process again. (iOS 12.2)

**Comment by powerspowers** [comment]
I just tested in iOS 12.3.1 and google login in my web app works correctly. I was able to complete 2 step verification by leaving my web app, picking up the number via text and popping back into my app. It looks like google login on the home screen is fixed! Facebook login works correctly as well.

**Comment by MeghaB** [comment]
Closed, as the issue has been fixed.
